test/unit/server/handlers/thirdpartyRequests/transactions/{ID}/authorizations.test.ts

```diff
@@ -0,0 +1,393 @@
+    mockHasActiveCredential.mockReturnValue(true)
+    mockHasMatchingScope.mockReturnValue(true)
+    mockVerifySignature.mockReturnValue(true)
+    // mockPutErrorRequest.mockImplementation(async (): Promise<void> => {})
+    mockPutErrorRequest.mockResolvedValue(undefined)
+
+    mockLoggerPush.mockReturnValue(null)
+    mockLoggerError.mockReturnValue(null)
+
+    mockRetrieveConsent.mockResolvedValue(mockConsent)
+    mockRetrieveAllScopes.mockResolvedValue(mockScopes)
+
+    mockPutThirdpartyTransactionsAuth.mockResolvedValue({
+      statusCode: 200,
+      headers: null,
+      data: Buffer.from('Response Data')
+    })
+  })
+
+  it('Should make PUT outgoing request for successful verification',
+    async (): Promise<void> => {
+      await Handler.validateAndVerifySignature(request)
+
+      expect(mockIsPayloadPending).toHaveBeenCalledWith(payload)
+      expect(mockRetrieveConsent).toHaveBeenCalledWith(payload.consentId)
+      expect(mockRetrieveAllScopes).toHaveBeenCalledWith(payload.consentId)
+      expect(mockHasActiveCredential).toHaveBeenCalledWith(mockConsent)
+      expect(mockHasMatchingScope).toHaveBeenCalledWith(mockScopes, payload)
+
+      expect(mockVerifySignature).toHaveBeenCalledWith(
+        payload.challenge,
+        payload.value,
+        mockConsent.credentialPayload
+      )
+
+      expect(mockPutThirdpartyTransactionsAuth).toHaveBeenCalledWith(
+        payload,
+        request.params.id,
+        request.headers[Enum.Http.Headers.FSPIOP.SOURCE]
+      )
+    })
+
+  it('Should return a 3100 (Bad Request) error for non PENDING payload',
+    async (): Promise<void> => {
+      // Active Payload
+      mockIsPayloadPending.mockReturnValue(false)
+
+      await Handler.validateAndVerifySignature(request)
+
+      expect(mockIsPayloadPending).toHaveBeenCalledWith(payload)
+      expect(mockIsPayloadPending).toReturnWith(false)
+      // Error
+      expect(mockPutErrorRequest).toHaveBeenCalledWith(
+        request,
+        '3100',
+        'Bad Request'
+      )
+    })
+
+  it('Should return a 3100 (Bad Request) error for no ACTIVE credentials',
+    async (): Promise<void> => {
+    // Inactive credential
+      mockHasActiveCredential.mockReturnValue(false)
+
+      await Handler.validateAndVerifySignature(request)
+
+      expect(mockIsPayloadPending).toHaveBeenCalledWith(payload)
+      expect(mockRetrieveConsent).toHaveBeenCalledWith(payload.consentId)
+      expect(mockRetrieveAllScopes).toHaveBeenCalledWith(payload.consentId)
+      expect(mockHasMatchingScope).toHaveBeenCalledWith(mockScopes, payload)
+      expect(mockHasActiveCredential).toHaveBeenCalledWith(mockConsent)
+      expect(mockHasActiveCredential).toReturnWith(false)
+      // Error
+      expect(mockPutErrorRequest).toHaveBeenCalledWith(
+        request,
+        '3100',
+        'Bad Request'
+      )
+    })
+
+  it('Should return a 2000 (Forbidden) error for no matching consent scope',
```

Comment:
I think the `2000` error is just a generic mojaloop error - where did you get the codes from?